third-party/realdds/src/dds-notification-server.cpp

```diff
@@ -0,0 +1,138 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include <realdds/dds-notification-server.h>
+
+#include <realdds/dds-participant.h>
+#include <realdds/dds-publisher.h>
+#include <realdds/dds-stream-server.h>
+#include <realdds/dds-utilities.h>
+#include <realdds/topics/dds-topics.h>
+#include <realdds/dds-topic.h>
+#include <realdds/dds-topic-writer.h>
+
+#include <fastdds/dds/topic/Topic.hpp>
+
+#include <fastdds/dds/domain/DomainParticipant.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/DataWriterListener.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <librealsense2/utilities/concurrency/concurrency.h>
+
+
+using namespace eprosima::fastdds::dds;
+
+namespace realdds {
+
+
+dds_notification_server::dds_notification_server( std::shared_ptr< dds_publisher > const & publisher, const std::string & topic_name )
+    : _publisher( publisher )
+    , _notifications_loop( [this]( dispatcher::cancellable_timer timer ) {
+        if( _active )
```

Comment:
Please move lambda logic to a named function, like `notification_loop_logic` or `wait_to_send_messages`.

Such long lambdas, and in a constructor initialization list, are hard to read, search and hence maintain. Lets try to keep our code [CLEAN](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29)er

Reply:
In another PR -- it's not so easy to move a lambda, you have to create a whole structure.
Don't forget, this isn't new code.